js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx

```diff
@@ -387,27 +391,53 @@ export const ConcurrencyLimits = ({
         concurrencyKey={selectedKey}
         onUpdate={refetch}
       />
-    </>
+    </Box>
   );
 };
 
-const ConcurrencyLimitHeader = ({onAdd}: {onAdd?: () => void}) => (
-  <Box
-    flex={{justifyContent: 'space-between', alignItems: 'center'}}
-    padding={{vertical: 16, horizontal: 24}}
-    border="top-and-bottom"
-  >
-    <Box flex={{alignItems: 'center', direction: 'row', gap: 8}}>
-      <Subheading>Global op/asset concurrency</Subheading>
-      <Tag>Experimental</Tag>
+const ConcurrencyLimitHeader = ({
+  onAdd,
+  setSearch,
+  search,
+}: {
+  onAdd?: () => void;
+} & (
+  | {
+      setSearch: (searchString: string) => void;
+      search: string;
+    }
+  | {setSearch?: never; search?: never}
+)) => {
```

Comment:
nit: probably just something i am not used to but this way of describing types is hard to parse to me

Reply:
This is to be specific about what are the valid parameter combinations.

Reply:
(eg. `search` and `setSearch` must both be set, or neither be set, you can't specify just one of them).